.github/workflows/test.yml

```diff
@@ -27,26 +27,33 @@ jobs:
         shell: bash
       # Setup the environment for the tests
       - name: Install stack
-        run: ghcup install stack recommended
+        run: |
+          export GHCUP_INSTALL_BASE_PREFIX=$(pwd)/test-workspace/bin
+          ghcup install stack latest
+          ghcup install stack recommended
+          ghcup list -t stack -c installed -r
+        shell: bash
       - name: Install cabal
-        run: ghcup install cabal recommended
+        run: |
+          export GHCUP_INSTALL_BASE_PREFIX=$(pwd)/test-workspace/bin
+          ghcup install cabal recommended
+          ghcup install cabal latest
+          ghcup list -t cabal -c installed -r
+        shell: bash
       - name: Install GHC
         run: |
+          export GHCUP_INSTALL_BASE_PREFIX=$(pwd)/test-workspace/bin
           ghcup install ghc ${{matrix.ghc}}
-          ghcup set ghc ${{matrix.ghc}}
```

Comment:
why remove setting ghc?

Reply:
I just remove this for testing, looks it has no effect.